vllm/entrypoints/openai/api_server.py

```diff
@@ -116,7 +116,14 @@ async def build_async_engine_client(args) -> AsyncIterator[AsyncEngineClient]:
 
         # Build RPCClient, which conforms to AsyncEngineClient Protocol.
         async_engine_client = AsyncEngineRPCClient(port)
-        await async_engine_client.setup()
+
+        while True:
+            try:
+                await async_engine_client.setup()
+                break
+            except TimeoutError as e:
+                if not rpc_server_process.is_alive():
+                    raise RuntimeError("Server crashed") from e
```

Comment:
Would be best to move this within the `try` block below I think

Reply:
Right, so that the clean up code in the finally section is called. Just moving the code into the try block caused another "hang" because it turns out that `context.destroy()` by default waits for 30 seconds if there are undelivered messages, which is the case if the server process crashed. So I added `linger=0` to the socket.close() calls to prevent that. I think this makes sense because in all uses of `self.socket()` the client sends a request and waits for the answer, so in all these cases we know that the server has seen the message.